File: webapp/src/lib/inbox.ts
// In-app inbox — job-alert emails forwarded to the user's account.
import { api } from "./api";

export interface InboxMessage {
  id: string;
  source: string;
  sender: string;
  subject: string;
  snippet: string;
  job_ids: string[];
  ingested: number;
  is_read: boolean;
  received_at: string;
}

export function getInboxAddress() {
  return api<{ address: string }>("/api/v1/inbox/address");
}
export function listInbox() {
  return api<InboxMessage[]>("/api/v1/inbox");
}
export function markInboxRead(id: string) {
  return api<InboxMessage>(`/api/v1/inbox/${id}/read`, { method: "POST", body: {} });
}
export function deleteInboxMessage(id: string) {
  return api<void>(`/api/v1/inbox/${id}`, { method: "DELETE" });
}
export function syncGmail() {
  return api<{ fetched: number; ingested: number; message_ids: string[] }>("/api/v1/inbox/sync-gmail", { method: "POST", body: {} });
}

File: src/jobsearch/models/inbox.py
"""In-app inbox — job-alert emails the user forwards to their account."""

from __future__ import annotations

from datetime import datetime

from pydantic import Field

from jobsearch.models.common import DomainModel, new_id, utcnow


class InboxMessage(DomainModel):
    id: str = Field(default_factory=lambda: new_id("inmsg_"))
    user_id: str
    source: str = ""  # detected board (linkedin/indeed/…)
    sender: str = ""  # From header
    subject: str = ""
    snippet: str = ""  # short preview
    job_ids: list[str] = Field(default_factory=list)  # postings ingested from this email
    ingested: int = 0
    is_read: bool = False
    received_at: datetime = Field(default_factory=utcnow)
